crates/jstzd/src/task/octez_client.rs

```diff
@@ -0,0 +1,199 @@
+                .endpoint
+                .unwrap_or(Endpoint::try_from(Uri::from_str(&node_default_endpoint)?)?),
+            disable_unsafe_disclaimer: self.disable_unsafe_disclaimer,
+        })
+    }
+    fn validate_binary_path(&self) -> Result<()> {
+        if let Some(binary_path) = &self.binary_path {
+            if !binary_path.exists() {
+                bail!("Binary path does not exist");
+            }
+            if !binary_path.is_file() {
+                bail!("Binary path is not a file");
+            }
+        }
+        Ok(())
+    }
+    fn validate_base_dir(&self) -> Result<()> {
+        if let Some(base_dir) = &self.base_dir {
+            if !base_dir.exists() {
+                bail!("Base directory does not exist");
+            }
+            if !base_dir.is_dir() {
+                bail!("Base directory is not a directory");
+            }
+        }
+        Ok(())
+    }
+}
+
+// TODO: remove later
+#[allow(dead_code)]
+enum Directory {
+    TempDir(TempDir),
+    Path(PathBuf),
+}
+
+// TODO: remove later
+#[allow(dead_code)]
+pub struct OctezClient {
+    binary_path: PathBuf,
+    base_dir: Directory,
+    endpoint: Endpoint,
+    disable_unsafe_disclaimer: bool,
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    #[test]
+    fn builds_default_octez_client() {
+        let octez_client = OctezClientBuilder::new().build().unwrap();
+        assert_eq!(
+            octez_client.binary_path.to_str().unwrap(),
+            DEFAULT_BINARY_PATH
+        );
+        assert!(matches!(octez_client.base_dir, Directory::TempDir(_)));
+        assert!(!octez_client.disable_unsafe_disclaimer);
+        assert_eq!(octez_client.endpoint, Endpoint::localhost(8732))
+    }
+
+    #[test]
+    fn temp_dir_is_created_by_default() {
+        let octez_client = OctezClientBuilder::new().build().unwrap();
+        match octez_client.base_dir {
+            Directory::TempDir(temp_dir) => assert!(temp_dir.path().exists()),
+            _ => panic!("Expected TempDir"),
+        };
+    }
+
+    #[test]
+    fn temp_dir_is_removed_on_drop() {
+        let octez_client = OctezClientBuilder::new().build().unwrap();
+        match &octez_client.base_dir {
+            Directory::TempDir(temp_dir) => {
+                let temp_dir_path = temp_dir.path().to_path_buf();
+                drop(octez_client);
+                assert!(!temp_dir_path.exists());
+            }
+            _ => panic!("Expected TempDir"),
+        };
+    }
+
+    #[test]
+    fn sets_custom_binary_and_dir_path() {
+        let binary_path = PathBuf::from("/tmp/binary1");
+        let dir_path = PathBuf::from("/tmp/dir1");
+        let _ = std::fs::remove_dir(&binary_path);
+        let _ = std::fs::remove_file(&binary_path);
+        let _ = std::fs::remove_dir(&dir_path);
+        let _ = std::fs::remove_file(&dir_path);
+        std::fs::File::create(&binary_path).unwrap();
+        std::fs::create_dir(&dir_path).unwrap();
```

Comment:
To avoid managing the temp files, use `NamedTempFile` then get its path with `NamedTempFile::path`. The same can be done `TempDir::new` and `TempDir::path`

Reply:
much cleaner 🧹